v-next/core/src/internal/hre.ts

```diff
@@ -37,9 +37,18 @@ export class HardhatRuntimeEnvironmentImplementation
     // TODO: Or maybe don't clone at all
     const clonedUserConfig = inputUserConfig;
 
+    // TODO: When all loads where based on a hardhat config file
+    // on disk we used the config file's directory as the base path
+    // We need to decide if the current working directory is the right
+    // option for module resolution now
+    const basePathForNpmResolution = process.cwd();
```

Comment:
I pulled up the point at which the npm resolution path is set. Currently this is set to `process.cwd()`, but that should probably change to `process.cwd()` for programmatically loaded config, and the directory of the `Hardhat.config.{js,ts}` if loaded from the command line.

Reply:
agreed